### DISCUSSION TITLE
Recording 15min per hour between 8AM to 6PM for selective days

### DISCUSSION THREAD

**Discussion by DavidBarriere** [gopro/labs#723] [Feature Requests]
![Dyade_01_AcquisitionScript-1_17_2024, 23 34 36](https://github.com/gopro/labs/assets/101575749/7c3343c9-140f-4240-b1a3-8d1f5b353ee9)

Hi guys.
Thank you for the latest answer : I reached to find a way to record 15min videos each hour all day long.
But I would like to be more selective to avoid data saturation and choose time for acquisition (ex. between 8AM to 6PM) and days of recording (based on calendar days (1st to 31).
I wrote this code but it behave curiously taking short videos. It's like the gopro start and stop acquisitions iteratively as an infernal loop.
Do you have any idea why ??

Additionally, The code is heavy to ready because of the lack of a simple equal option in the variable/sensor block (>= and < are available but not more) resulting an inflation of condition just to set an acquisition day.
Example "if variable/sensor_J >= 22 and if variable/sensor_J < 23 do somethings" 
to select only the day 22 of each month. 

So in my code 8 variables (S T U V W X Y Z) are necessary to select specific days through a month that I am interested in.
Do you have any way to get a visual script more readable, or did you planed to add an equal function within the variable/sensor block (it should be more easier to fix all settings available).
Thanks for your job : it's help a lot.

David

**Comment by dnewman-gpsw** [comment]
That looks very complicated.  You seem to be setting r27p30fW multiple times (not even a mode on 10,11 or HERO12),so that seems unnecessary.  Set once and the preset will remember. == conditions are planned for the next release.

but you don't need == to make this easier.

Here I used a bit mask to select the record days:
![Capture with day Mask for 15mins_hr in time range-1_17_2024, 16_31_03](https://github.com/gopro/labs/assets/10856084/83942667-a0de-440a-9b1f-b5b21a4ae295)

**Comment by DavidBarriere** [comment]
Thank you for the rapid reply.
I Tried to set a specific acquisition day and a specific time to start shooting following the method but it doesn't work.
![Camera_02_Script_02-1_18_2024, 14 10 19](https://github.com/gopro/labs/assets/101575749/73b1bb0b-74c3-42c3-8172-7e469028cfd7)

I really confused on why.

D

**Comment by dnewman-gpsw** [comment]
I see a couple bugs. You have mixed mixed up date-day (1-31) with day of the week (0-6).  Add you do a bitwise mask on minute, which where perfect as it was. I can upgrade my version as it was cleaner.  I hope you know the QR Code PNG can be imported into Blockly via Import Image.

Script with comments:
![image](https://github.com/gopro/labs/assets/10856084/687aa805-4367-4cec-bded-ea4f71ad09af)


![Capture with a day Mask for 15mins an hour between 8am and 6pm-1_18_2024, 12_46_59](https://github.com/gopro/labs/assets/10856084/1f5bcb92-0b4d-426d-b274-94d13d41f372)

The mask is set to 127, so everyday to get you started.

**Reply by DavidBarriere** [reply]
Okay thank you.
But the thing that I do not catch is the 127 in the variable M. Is it the 127th day from January 1st (in this case the day 01/01 is set at 0 or 1)?
In this code the mask provided does it an implicit (1) or an explicit mask (0).

"I hope you know the QR Code PNG can be imported into Blockly via Import Image."
Yes and untill now it was working, but I failed to load this QRcode :(

Thank you again.

D

**Reply by dnewman-gpsw** [reply]
Mask M is a bitwise mask.

1 = Sun
2 = Mon
4 = Tues
8 = Wed
16 = Thurs
32 = Fri
64 = Sat

127 = 1+2+4+8+16+32+64  All days of the week.

If you want Mon, Wed and Fri. Mask M = 2+8+32 = 42

I just tested, the PNG about loads correctly.  Reflesh your browser, it was updated yesterday.

**Reply by DavidBarriere** [reply]
It sounds working perfectly! Thank you so much.

**Comment by DavidBarriere** [comment]
Hi guys.
Thank you again for the latest answer.
I am still questioning regarding the command "wifi join network".

What does this command really do ?

Does it open the ble and the wifi is reacheable or does it still open the wifi to allow to connect the camera to a computer (for example) ? 

If the command still the wifi open is it the 5gHz or the 2.4 ghz ?

Otherwise there is a way to maintain all night long the ble or the wifi open to download the data acquired during the day ?

Thank you

David

**Reply by dnewman-gpsw** [reply]
It only does the WiFi network connection using previous settings (5 vs 2.4Ghz).  Nothing to do with BLE.  
You can't download directly, only the !U command to upload to GoPro cloud is supported.

On HERO12 you can do !U on it own as it uses the credentials you setup with Quik. The join network is not required on HERO12.  For livestreaming !W is required on all models.

BLE shutsdown automatically after a few hours of the camera off.  If your camera is on the BLE should be up, and can be enabled with oW1 (think, maybe have change with later models.)

**Reply by DavidBarriere** [reply]
Okay I see.
Bassically, I want to maintain the wifi open all night long to download data from a raspi using http command from open gopro.

To relief the procedure I wished to add a condition to the qrCode to maintain the wifi open for example between 5PM to 9PM. During this period I can program the raspi to connect the gopro by wifi and send downloading and clean sd card http commands.
 
To maintain the wifi open I added something like this to the previous qRCode: 
if HOUR > 5PM
   if HOUR < 9AM
       "wifi join network".
   end
end

But I failed to download data.
Shutdown is set as never, wifi band at 2.4GHz and the gopro is appaired to the raspy.

I think it's because the command open and close the wifi iteratively when the command is placed within the loop. There is a way to program a QRCode which maintain the wifi open for a specific time period ?

**Reply by DavidBarriere** [reply]
![Test_Acquisition_9h-17h_HH 30-HH 40+WIFI_Connexion-1_31_2024, 16 37 46](https://github.com/gopro/labs/assets/101575749/61c840bf-5472-4bca-b6ea-69db245adf00)

In Summary I tested this code.

**Reply by dnewman-gpsw** [reply]
I've never used Open GoPro, so this is out of my area.  There is a request to get a bridge from Open GoPro to Labs firmware, mainly so Labs commands can be issued through Open GoPro connections.  The !W command will join wifi at that point, but it might timeout.  Start simpler. You should first determine if Open GoPro will use the connection !W opens (requiring a previous setup of !MJOIN="ssid:password"), as it was added for livestreaming, and not tested for other applications.

**Reply by DavidBarriere** [reply]
Okay I see.

Thank you!
